rmf_task/include/rmf_task/TaskPlanner.hpp

```diff
@@ -39,6 +39,73 @@ class TaskPlanner
 {
 public:
 
+  /// The strategy to use when assigning tasks to robots during node expansion.
+  /// This struct defines the various profiles and their associated weights
+  /// for cost calculation.
+  struct TaskAssignmentStrategy
+  {
+    enum class Profile
+    {
+      DefaultFastest,
+      BatteryAware,
+      Custom
```

Comment:
I'm deliberating whether `Custom` is expressing the right idea here. I think it would be more accurate to say `Unset`, since that makes it more explicit that the user needs to follow up and set all the relevant weights explicitly.